### PR TITLE
Vault-5619: Transit BYOK Documentation

### PR DESCRIPTION
**Pull request by rculpepper**
Docs changes for Transit BYOK API endpoints and description for constructing the ciphertext for the key

### CODE REVIEW

File: website/content/docs/secrets/transit.mdx [hunk 1]
```diff
@@ -232,6 +232,57 @@ the proper permission, it can use this secrets engine.
     data, since the process would not be able to get access to the plaintext
     data.
 
+## BYOK
```
Comment (sgmiller) [inline website/content/docs/secrets/transit.mdx:235]: Lets be explicit here: "Bring Your Own Key (BYOK)"

Comment (sgmiller) [reply website/content/docs/secrets/transit.mdx:235]: Also, I'd like some language discouraging BYOK when not needed.  Something like "While creating a key within the Transit Secrets Engine and letting Vault create and manage the key material is more secure, there may be a need to bring a key from an existing HSM or system.  The key import functionality enables this case."

**Review by sgmiller — CHANGES_REQUESTED** [review]
Just a couple notes.

File: website/content/docs/secrets/transit.mdx [hunk 2]
```diff
@@ -232,6 +232,57 @@ the proper permission, it can use this secrets engine.
...
+### HSM
+
+If the key is being imported from an HSM that supports PKCS#11, there are
+two possible scenarios:
+
+- If the HSM supports the CKM_RSA_AES_KEY_WRAP mechanism, that can be used to wrap the
+target key using the wrapping key.
+
+- Otherwise, two mechanisms can be combined to wrap the target key. First, an AES key should
+be generated and then used to wrap the target key using the CKM_AES_KEY_WRAP_PAD mechanism.
```
Comment (cipherboy) [inline website/content/docs/secrets/transit.mdx:260]: Just a note, this mechanism is wrong. Per the [updated PKCS#11 v3.0 spec](http://docs.oasis-open.org/pkcs11/pkcs11-curr/v3.0/os/pkcs11-curr-v3.0-os.html#_Toc370634387) and the Transit implementation, this should be `CKM_AES_KEY_WRAP_KWP`.

**Review by cipherboy — CHANGES_REQUESTED** [review]

**Pull request closed without merging**